src/org/authInfo.ts

```diff
@@ -1065,6 +1070,30 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
     }
     throw new SfError(errorMsg);
   }
+
+  /**
+   * Returns `true` if the org is a Dev Hub.
+   *
+   * Check access to the ScratchOrgInfo object to determine if the org is a dev hub.
+   */
+  private async determineIfDevHub(instanceUrl: string, accessToken: string): Promise<boolean> {
```

Comment:
@cristiand391 The [Org class already has a function to determine if an org is a devhub](https://github.com/forcedotcom/sfdx-core/blob/cd/detect-devhub/src/org/org.ts#:~:text=public%20async-,determineIfDevHubOrg,-(forceServerCheck%20%3D%20false)%3A%20Promise) Is there a reason it is not being used here?

Reply:
Yes, we can't use it here because it writes the authFile: https://github.com/forcedotcom/sfdx-core/blob/d0c02854ba1e80184589b1eb4b42ceb3a3ef0ee0/src/org/org.ts#L425

For example, let's say a consumer creates a new instance of `AuthInfo`:

```ts
import { Connection,AuthInfo } from '@salesforce/core'

(async() => {
  const user = await AuthInfo.create({
    username: 'cris@test.org',
    oauth2Options: {
      clientId: '1234',
      privateKeyFile: '/home/cris.key'
    }
  })

  user.getUsername()
})()
```

after that code is executed, you should not see an auth file in `~/.sfdx | ~/.sf/sf.json` because you never called `user.save()`.

Both `Org` and `Connection` classes write the auth instance to a file.

Reply:
Is there any code in common between the two that can be extracted into a distinct function?

Reply:
I don't think so, Steve about this in Slack too, to sum up:

1) This new method in `AuthInfo` is private, Org exposes `determineIfDevHub` as a public method.
2) While we could remove the server check code from [`Org.determineIfDevHub()`](https://github.com/forcedotcom/sfdx-core/blob/d0c02854ba1e80184589b1eb4b42ceb3a3ef0ee0/src/org/org.ts#L414), already created auth files may not have `isDevHub` set and not doing the server check in `Org` could break code like this: https://github.com/forcedotcom/sfdx-core/blob/fa6e4e25255df57f2b80d05bfc62c38d192aa924/src/org/scratchOrgInfoApi.ts#L197

So we want to keep the code in `Org` to handle old auth files, for newly created auth files Org will still check the cached value first.